Bugfix and Change

Bletch1971 opened the pull request:
1. Add a fix for the Message of the Day to output multiple lines correctly.
2. SystemIniFile.cs - added Multiline Attribute and changed Serialize and
   Deserialize to handle NewLine characters
3. ServerProfile.cs - added Multiline = true to the MOTD property
4. Added two Max XP reset buttons, one for player section, one for dino section.
5. en-US.xaml - added generic tooltip for the button
6. ServerProfile.cs - added 2 constants values for MaxXP, updated
   DependancyProperties to use constant values and added 2 new methods for
   resetting the values.
7. ServerSettingsControl.xaml - added two new reset buttons to the player
   and dino section next to the Max XP slider. Also added margins to the
   other sliders to align them.

Bletch1971 committed:
Bugfix and Change

1. Add a fix for the Message of the Day to output multiple lines
correctly.
* SystemIniFile.cs - added Multiline Attribute and changed Serialize and
Deserialize to handle NewLine characters
* ServerProfile.cs - added Multiline = true to the MOTD property

2. Added two Max XP reset buttons, one for player section, one for dino
section.
* en-US.xaml - added generic tooltip for the button
* ServerProfile.cs - added 2 constants values for MaxXP, updated
DependancyProperties to use constant values and added 2 new methods for
resetting the values.
* ServerSettingsControl.xaml - added two new reset buttons to the player
and dino section next to the Max XP slider. Also added margins to the
other sliders to align them.

Bletch1971 commented:
This is what the reset buttons look like

![playerreset](https://cloud.githubusercontent.com/assets/4722854/11522872/3e1e2154-9908-11e5-9283-15f4da74ba4d.PNG)

![dinoreset](https://cloud.githubusercontent.com/assets/4722854/11522875/46c09a44-9908-11e5-85d2-eb5ec9fb4532.PNG)

ChronosWS commented on ARK Server Manager/Globalization/en-US/en-US.xaml:
```diff
@@ -290,6 +290,7 @@
     <sys:String x:Key="ServerSettings_MaxXPLabel">Max XP:</sys:String>
     <sys:String x:Key="ServerSettings_MaxXPTooltip">Set the maximum XP cap.</sys:String>
     <sys:String x:Key="ServerSettings_MaxXPUnits">xp</sys:String>
+    <sys:String x:Key="ServerSettings_ResetMaxXP">Reset Max XP Cap to official default</sys:String>
```
*Tooltip

ChronosWS commented on ARK Server Manager/Lib/Model/ServerProfile.cs:
```diff
@@ -25,7 +25,10 @@ public interface ISettingsBag
     public class ServerProfile : DependencyObject
-    { 
+    {
+        private const int DEFAULT_MAX_EXPERIENCE_POINTS_DINO = 275000;
```
These should go in GameData and can be public.

Bletch1971 committed:
Pull Request Corrections

1. Changed localisation string to include Tooltip and altered tooltip
text to be consistent with previous tooltip.
2. Moved Default XP constants to GameData and made public.
3. Changed ServerProfile to use GameData constants.
4. Changed tooltip localisation reference in ServerSettingsControl.

Bletch1971 commented:
Corrections made, happy to change anything else if needed.

Bletch1971 commented:
I would like to add a reset button to most of the sections, to reset that sections values, but it would mean a lot of new constant values being added to GameData. The reset button would be at the top of the section and resets all values in that section.

Are you OK with this?

ChronosWS commented:
I don't have a problem with it.  Is there a defaults file that comes with
the server now though?

On Wed, Dec 2, 2015 at 7:00 PM, Brett notifications@github.com wrote:

> I would like to add a reset button to most of the sections, to reset that
> sections values, but it would mean a lot of new constant values being added
> to GameData. The reset button would be at the top of the section and resets
> all values in that section.
> 
> Are you OK with this?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ChronosWS/ARK-Dedicated-Server-Tool/pull/39#issuecomment-161501901
> .

Bletch1971 commented:
No defaults file that I know of, I was just going to reset the section values back to the defaults you use when you create a new profile.
I have tried in the past to get default values for all the server settings used by ARK, but nothing is ever complete or explained properly on Steam or the wiki.
Getting server setting information is painful.
